### PR TITLE
Add dto logic + Tests, add GetAll to Frontend

### PR DESCRIPTION
**Pull request by fknoerzer**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/calculations/EmissionCalculationService.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package de.neuefische.backend.calculations;
+
+import de.neuefische.backend.dto.TripDto;
+
```
Comment (Gadenko) [inline backend/src/main/java/de/neuefische/backend/calculations/EmissionCalculationService.java:4]:
```suggestion
```

File: backend/src/main/java/de/neuefische/backend/controller/TripController.java [hunk 2]
```diff
@@ -1,7 +1,9 @@
 package de.neuefische.backend.controller;
 
+import de.neuefische.backend.dto.TripDto;
 import de.neuefische.backend.model.Trip;
 import de.neuefische.backend.service.TripService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
```
Comment (Gadenko) [inline backend/src/main/java/de/neuefische/backend/controller/TripController.java:8]:
```suggestion
```

File: backend/src/main/java/de/neuefische/backend/dto/TripDto.java [hunk 3]
```diff
@@ -5,23 +5,32 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
```
Comment (Gadenko) [inline backend/src/main/java/de/neuefische/backend/dto/TripDto.java:7]:
```suggestion
```

File: backend/src/main/java/de/neuefische/backend/model/Trip.java [hunk 4]
```diff
@@ -1,13 +1,14 @@
 package de.neuefische.backend.model;
 
+
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 import org.springframework.data.annotation.Id;
 import org.springframework.data.mongodb.core.mapping.Document;
```
Comment (Gadenko) [inline backend/src/main/java/de/neuefische/backend/model/Trip.java:10]:
```suggestion
```

File: backend/src/main/java/de/neuefische/backend/model/Trip.java [hunk 5]
```diff
@@ -18,15 +19,32 @@ public class Trip {
...
+    private double numberOfElectronicItems;
+    private double numberOfSouvenirItems;
+    private String customShoppingItem;
+    private double customShoppingItemEmission;
+    private double amountOfGolfRounds;
+    private double amountOfSkiingDays;
+    private double amountOfBeautyDays;
+    private String customActivity;
+    private double customActivityEmission;
+
```
Comment (Gadenko) [inline backend/src/main/java/de/neuefische/backend/model/Trip.java:43]:
```suggestion
```

File: backend/src/test/java/de/neuefische/backend/calculations/EmissionCalculationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
+package de.neuefische.backend.calculations;
+
+import de.neuefische.backend.dto.TripDto;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+
```
Comment (Gadenko) [inline backend/src/test/java/de/neuefische/backend/calculations/EmissionCalculationServiceTest.java:7]:
```suggestion
```

File: backend/src/test/java/de/neuefische/backend/controller/TripControllerTest.java [hunk 7]
```diff
@@ -1,20 +1,26 @@
...
 
+import de.neuefische.backend.dto.TripDto;
 import de.neuefische.backend.model.Trip;
 import de.neuefische.backend.repository.TripRepo;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
 import org.springframework.test.web.reactive.server.WebTestClient;
```
Comment (Gadenko) [inline backend/src/test/java/de/neuefische/backend/controller/TripControllerTest.java:12]:
```suggestion
```

File: backend/src/test/java/de/neuefische/backend/service/TripServiceTest.java [hunk 8]
```diff
@@ -1,9 +1,12 @@
 package de.neuefische.backend.service;
 
+import de.neuefische.backend.calculations.EmissionCalculationService;
+import de.neuefische.backend.dto.TripDto;
 import de.neuefische.backend.model.Trip;
 import de.neuefische.backend.repository.TripRepo;
 import org.junit.jupiter.api.Test;
```
Comment (Gadenko) [inline backend/src/test/java/de/neuefische/backend/service/TripServiceTest.java:8]:
```suggestion
```

File: backend/src/test/java/de/neuefische/backend/service/TripServiceTest.java [hunk 9]
```diff
@@ -1,9 +1,12 @@
...
 
+import java.time.LocalDate;
 import java.util.List;
```
Comment (Gadenko) [inline backend/src/test/java/de/neuefische/backend/service/TripServiceTest.java:11]:
```suggestion
```

File: frontend/src/model/Trip.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    typeOfTransports: string;
+    typeOfAccommodation: string;
+    typeOfDiet: string;
+    numberOfClothingItems: number;
+    numberOfElectronicItems: number;
+    numberOfSouvenirItems: number;
+    amountOfGolfRounds: number;
+    amountOfSkiingDays: number;
+    amountOfBeautyDays: number;
+
```
Comment (Gadenko) [inline frontend/src/model/Trip.ts:19]:
```suggestion
```

File: backend/src/main/java/de/neuefische/backend/controller/TripController.java (hunk 2 above)
Comment (fknoerzer) [reply backend/src/main/java/de/neuefische/backend/controller/TripController.java:8]: Fixed

File: backend/src/main/java/de/neuefische/backend/dto/TripDto.java (hunk 3 above)
Comment (fknoerzer) [reply backend/src/main/java/de/neuefische/backend/dto/TripDto.java:7]: Fixed

File: backend/src/main/java/de/neuefische/backend/model/Trip.java (hunk 4 above)
Comment (fknoerzer) [reply backend/src/main/java/de/neuefische/backend/model/Trip.java:10]: Fixed

File: backend/src/main/java/de/neuefische/backend/model/Trip.java (hunk 5 above)
Comment (fknoerzer) [reply backend/src/main/java/de/neuefische/backend/model/Trip.java:43]: Fixed

**Pull request merged**